scripts/build/mirror.ts

```diff
@@ -89,24 +90,24 @@ export const getMatchingBrowserVersion = (
     );
   }
 
-  for (const r of releaseKeys) {
-    const release = browserData.releases[r];
+  let firstRelease = true;
+  for (const v of versions) {
+    const release = browserData.releases[v];
     if (
-      ['chrome', 'chrome_android'].includes(browserData.upstream) &&
-      targetBrowser !== 'chrome_android' &&
-      release.engine == 'Blink' &&
-      sourceRelease.engine == 'WebKit'
+      // Handle Chromium forks when upstream version is pre-Blink
+      (['chrome', 'chrome_android'].includes(browserData.upstream) &&
+        targetBrowser !== 'chrome_android' &&
+        release.engine == 'Blink' &&
+        sourceRelease.engine == 'WebKit') ||
+      // Match same or newer engine version
+      (release.engine == sourceRelease.engine &&
+        release.engine_version &&
+        sourceRelease.engine_version &&
+        compare(release.engine_version, sourceRelease.engine_version, '>='))
     ) {
-      // Handle mirroring for Chromium forks when upstream version is pre-Blink
-      return range ? `≤${r}` : r;
-    } else if (
-      release.engine == sourceRelease.engine &&
-      release.engine_version &&
-      sourceRelease.engine_version &&
-      compare(release.engine_version, sourceRelease.engine_version, '>=')
-    ) {
-      return r;
```

Comment:
This is the key line. Previously ranges were never propagated here, and now they are except for the first release.